perses/rjmc/geometry.py

```diff
@@ -293,6 +293,8 @@ def _logp_propose(self, top_proposal, old_positions, beta, new_positions=None, d
             else:
                 if direction == 'forward':
                     constraint = self._get_bond_constraint(atom, bond_atom, top_proposal.new_system)
+                    if constraint is None:
```

Comment:
Maybe also add a case "`if type(constraint) == float/float64`" or something and an else case.